pkg/controller/controlplane/genericactuator/actuator.go

```diff
@@ -388,11 +389,20 @@ func (a *actuator) deleteControlPlane(
 	if err := extensionscontroller.DeleteManagedResource(ctx, a.client, cp.Namespace, storageClassesChartResourceName); err != nil {
 		return errors.Wrapf(err, "could not delete managed resource containing storage classes chart for controlplane '%s'", util.ObjectName(cp))
 	}
-
 	if err := extensionscontroller.DeleteManagedResource(ctx, a.client, cp.Namespace, controlPlaneShootChartResourceName); err != nil {
 		return errors.Wrapf(err, "could not delete managed resource containing shoot chart for controlplane '%s'", util.ObjectName(cp))
 	}
 
+	timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+
+	if err := extensionscontroller.WaitUntilManagedResourceDeleted(timeoutCtx, a.client, cp.Namespace, storageClassesChartResourceName); err != nil {
+		return errors.Wrapf(err, "error while waiting for managed resource containing storage classes chart for controlplane '%s' to be deleted", util.ObjectName(cp))
+	}
+	if err := extensionscontroller.WaitUntilManagedResourceDeleted(timeoutCtx, a.client, cp.Namespace, controlPlaneShootChartResourceName); err != nil {
```

Comment:
I don't think it's a good idea to reuse `timeoutCtx` since `WaitUntilManagedResourceDeleted` blocks and quite some time has already passed from the previous wait. You could either wait in parallel or use a new timeout context.